### PR TITLE
Fix more issues with migration 23

### PR DESCRIPTION
**Pull request by ta264**
#### Database Migration
NO

#### Description
Fixes a couple more issues that have shown up in sentry:
1. where `CurrentRelease` is completely null
1. where we have duplicate `ForeignReleaseId` (the previous fix would trip up when trying to assign `ReleaseId` to each `Track`

In 1., I have put in a dummy release so that the migration succeeds and the first medium's tracks will show up in the UI.  This will get deleted and re-created by the metadata refresh (as will all the associated tracks and mapped trackfiles).  For 2. I have put in a fake `ForeignReleaseId` - we know the `AlbumId` is unique so I've used that.  Again, this means the release and any tracks/trackfiles will be deleted and recreated but I can't think of a way we can avoid that.

The second commit fixes up the metadata refreshes.  It makes sure that we deal with the case where we have an item in the database but linked against a different parent.  We were previously querying for items by their parent (i.e. the item had to have the correct `parentId`) so we could end up trying to insert an item that already exists under a different (outdated) parent.

The new version gets items by both `foreignId` and `parent` to make sure that we capture both cases.  Then make sure to delete any that should no longer be present (because they have been deleted or moved to a new parent) before inserting new ones.

I think this will also fix `LIDARR-1A` about unique constraint failing on metadata insert.

### CODE REVIEW

**Comment by ta264** [comment]
Sentry issue: [LIDARR-1A](https://sentry.io/lidarr/lidarr/issues/351278015/?referrer=github_plugin)

**Comment by ta264** [comment]
Sentry issue: [LIDARR-13W](https://sentry.io/lidarr/lidarr/issues/816157527/?referrer=github_plugin)

**Comment by ta264** [comment]
Sentry issue: [LIDARR-13V](https://sentry.io/lidarr/lidarr/issues/816152613/?referrer=github_plugin)

**Comment by Qstick** [comment]
Todo: Add test case as discussed.

**Comment by ta264** [comment]
I've added in some tests for the migration, focussed on possibly weird values for the existing `CurrentRelease` that is used to populate the `AlbumReleases` table.  Let me know if you can think of any other cases we should test also.

PS I checked and the tests do fail if you revert the improvements to the migration, so looks like they're workin correctly.

File: src/NzbDrone.Core/Music/RefreshTrackService.cs [hunk 1]
```diff
@@ -37,12 +37,17 @@ public void RefreshTrackInfo(Album album)
...
             {
-
-                var existingTracks = _trackService.GetTracksByForeignReleaseId(release.ForeignReleaseId);
+                var dupeFreeRemoteTracks = release.Tracks.Value.DistinctBy(m => new { m.ForeignTrackId, m.TrackNumber }).ToList();
+                
+                // Search both ways to make sure we properly deal with tracks that have been moved from one release to another
+                // as well as deleting any tracks that have been removed from a release.
+                // note that under normal circumstances, a track would be captured by both queries.
+                var existingTracksByRelease = _trackService.GetTracksByForeignReleaseId(release.ForeignReleaseId);
+                var existingTracksById = _trackService.GetTracksByForeignTrackIds(dupeFreeRemoteTracks.Select(x => x.ForeignTrackId).ToList());
```
Comment (Qstick) [inline src/NzbDrone.Core/Music/RefreshTrackService.cs]: I don't think tracks can move from one release to another like releases can in MB can they? You would have to delete the track on one release during edit and add a new track to the other release.

Comment (ta264) [reply src/NzbDrone.Core/Music/RefreshTrackService.cs]: I'm not sure, I've never edited anything on MB. But this [sentry issue](https://sentry.io/lidarr/lidarr/issues/351278015/?referrer=github_plugin) has lots of events where track foreign Id unique constraint is violated and I'm not sure how else that would happen.

**Pull request merged**